### ISSUE TITLE
Opaque error message when xcodebuild is unavailable

### ISSUE THREAD

**Issue by macdrevx** [realm/SwiftLint#3565]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

If `xcodebuild -version` returns

```
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```

swiftlint fails with 

```bash
$ swiftlint lint
Fatal error: Loading sourcekitd.framework/Versions/A/sourcekitd failed: file SourceKittenFramework/library_wrapper.swift, line 31
[1]    83461 trace trap  swiftlint lint
```

Could we add some kind of detection for this condition and emit a more useful error message?

##### Complete output when running SwiftLint, including the stack trace and command used

Crash report:
[swiftlint_2021-03-10-100023_H2WF637UQ6NY.crash.zip](https://github.com/realm/SwiftLint/files/6117259/swiftlint_2021-03-10-100023_H2WF637UQ6NY.crash.zip)

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.43.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

See https://github.com/mapbox/mapbox-maps-ios/tree/412981f0712d5f76d6ce41015d21e9fdb9cdb521

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.

Yes:

```
$ find . -name ".swiftlint.yml" 
./.swiftlint.yml
./Tests/.swiftlint.yml
./Sources/.swiftlint.yml
```

* Which Xcode version are you using (check `xcodebuild -version`)?

```
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

No

**Comment by jpsim** [comment]
I agree this error isn't great. It's coming from here by the way: https://github.com/jpsim/SourceKitten/blob/863680921bcc3ba9785c9584d4dd195f8597c7d8/Source/SourceKittenFramework/library_wrapper.swift#L31

We'd need to change that from a fatal error to a non-fatal error that can be caught and handled.

Or perform validation up front before trying to load libraries that may not be available.

**Comment by htmlprogrammist** [comment]
Same problem here. Installation using homebrew

**Comment by gpoitch** [comment]
`sudo xcode-select -switch /Applications/Xcode.app/Contents/Developer` then it works

**Comment by micHar** [comment]
Keep in mind that there are also other options to fix the xcode-select issue, be sure to check them out before running `sudo xcode-select -switch...`. https://stackoverflow.com/questions/17980759/xcode-select-active-developer-directory-error

**Comment by duyhungtnn** [comment]
I was facing this recently , here is solution https://github.com/realm/SwiftLint/issues/1466#issuecomment-297265863
